recipes/botan/all/conanfile.py

```diff
@@ -0,0 +1,305 @@
+
+    def requirements(self):
+        if self.options.with_bzip2:
+            self.requires("bzip2/1.0.6")
+        if self.options.with_openssl:
+            self.requires("openssl/1.0.2t")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+        if self.options.with_sqlite3:
+            self.requires("sqlite3/3.30.1")
+        if self.options.with_boost:
+            self.requires("boost/1.71.0")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "botan-" + self.version
+        os.rename(extracted_dir, "sources")
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        with tools.chdir('sources'):
+            self.run(self._configure_cmd)
+            self.run(self._make_cmd)
+
+    def package(self):
+        self.copy(pattern="license.txt", dst="licenses", src="sources")
+        with tools.chdir("sources"):
+            self.run(self._make_install_cmd)
+
+    def package_info(self):
+        if self.settings.compiler == 'Visual Studio':
+            self.cpp_info.libs.append('botan')
+        else:
+            self.cpp_info.libs.extend(['botan-2'])
+            if self.settings.os != 'Windows':
+                self.cpp_info.system_libs.append('dl')
+            if self.settings.os == 'Linux':
+                self.cpp_info.system_libs.append('rt')
+            if self.settings.os == 'Macos':
+                self.cpp_info.frameworks = ['Security', 'CoreFoundation']
+            if not self.options.shared:
+                self.cpp_info.system_libs.append('pthread')
+        if self.settings.os == "Windows":
+            self.cpp_info.system_libs.extend(["ws2_32", "Crypt32"])
+
+        self.cpp_info.libdirs = ['lib']
+        self.cpp_info.bindirs = ['bin']
+        self.cpp_info.includedirs = ['include/botan-2']
+
+    def _validate_compiler_settings(self):
+        compiler = self.settings.compiler
+        version = Version(self.settings.compiler.version.value)
+
+        if compiler == "Visual Studio" and version < "14":
+            raise ConanInvalidConfiguration("Botan doesn't support MSVC < 14")
+
+        elif compiler == "gcc" and version >= "5" and compiler.libcxx != "libstdc++11":
+            raise ConanInvalidConfiguration(
+                'Using Botan with GCC >= 5 on Linux requires "compiler.libcxx=libstdc++11"')
+
+        elif compiler == "clang" and compiler.libcxx not in ["libstdc++11", "libc++"]:
+            raise ConanInvalidConfiguration(
+                'Using Botan with Clang on Linux requires either "compiler.libcxx=libstdc++11" ' \
+                'or "compiler.libcxx=libc++"')
+
+    @property
+    def _is_mingw_windows(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc" and os.name == "nt"
```

Comment:
Why do you need the `os.name == "nt"` check? I think it is enough like this
```suggestion
        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
```

Reply:
Huh, great question. Looks unnecessary to me, but I'll try to investigate how it came into the original recipe.

Reply:
Hm, it can probably go. Except if @SSE4 still knows why it was [added more than a year ago](https://github.com/bincrafters/conan-botan/commit/e1a819e84c30e8ce0d8db4c6bd8e6fa858e63fe4) :/

Reply:
it's to exclude cross-compiling case (Linux -> Windows)
in general, you need `msys2`, `run_in_win_bash` or similar things only for Windows MinGW, not for Linux MinGW

Reply:
We can go without this for now. Hopefully, that will be improved with the new cross-building model. Thanks for commenting!